src/idl/tests/annotation.c

```diff
@@ -67,6 +67,124 @@ CU_Test(idl_annotation, id_member)
   CU_ASSERT_FATAL(idl_is_member(c));
   CU_ASSERT_EQUAL(c->id.annotation, IDL_ID);
   CU_ASSERT_EQUAL(c->id.value, 1);
+
+  idl_delete_pstate(pstate);
+}
+
+CU_Test(idl_annotation, optional)
+{
+  idl_retcode_t ret;
+  idl_pstate_t *pstate = NULL;
+  idl_struct_t *s;
+  idl_member_t *c;
+  idl_declarator_t *d;
+  idl_union_t *u;
+  idl_case_t *uc;
+  const char structstr[] = "struct s {\n"
+                           "  char c_none;\n"
+                           "  @optional(false) char c_false;\n"
+                           "  @optional char c_default;\n"
+                           "  @optional(true) char c_true;\n"
+                           "  @optional char c_1_default, c_2_default;\n"
+                           "};\n";
```

Comment:
Add test some cases for other member types, e.g. aggregated types, sequences, arrays?

Reply:
Will do.